typescript/cli/src/deploy/warp.ts

```diff
@@ -397,28 +401,31 @@ async function getWarpCoreConfig(
   const warpCoreConfig: WarpCoreConfig = { tokens: [] };
 
   // TODO: replace with warp read
-  const tokenMetadata = await HypERC20Deployer.deriveTokenMetadata(
-    params.context.multiProvider,
-    params.warpDeployConfig,
-  );
-  assert(
-    tokenMetadata && isTokenMetadata(tokenMetadata),
-    'Missing required token metadata',
-  );
-  const { decimals, symbol, name } = tokenMetadata;
-  assert(decimals, 'Missing decimals on token metadata');
+  const tokenMetadata: TokenMetadataMap =
+    await HypERC20Deployer.deriveTokenMetadata(
+      params.context.multiProvider,
+      params.warpDeployConfig,
+    );
+
+  for (const [, config] of Object.entries(tokenMetadata)) {
+    assert(
+      tokenMetadata && isTokenMetadata(config),
+      'Missing required token metadata',
+    );
+    assert(config.decimals, 'Missing decimals on token metadata');
```

Comment:
can you include the chain name in the error message? imagine having multiple errros like this , it would be helpful to know to which chain each one belongs to